src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/AndroidInstrumentationBase.cs

```diff
@@ -0,0 +1,198 @@
+        {
+        }
+
+        protected virtual IEnumerable<string> Tests
+        {
+            get
+            {
+                yield return Assembly.GetExecutingAssembly().Location;
+            }
+        }
+
+        public override void OnCreate(Bundle arguments)
+        {
+            base.OnCreate(arguments);
+
+            foreach (var key in arguments.KeySet())
+            {
+                string value = arguments.GetString(key);
+                if (!string.IsNullOrEmpty(value))
+                {
+                    bundleArguments.Add(key, value);
+                }
+            }
+
+            if (bundleArguments.ContainsKey(ResultsFileArgumentName) == false)
+            {
+                bundleArguments.Add(ResultsFileArgumentName, "TestResults.xml");
+            }
+
+            Start();
+        }
+
+        public override async void OnStart()
+        {
+            base.OnStart();
+
+            var bundle = new Bundle();
+
+            var entryPoint = new TestsEntryPoint(bundleArguments);
+            entryPoint.TestsCompleted += (sender, results) =>
+            {
+                var message =
+                    $"Tests run: {results.ExecutedTests} " +
+                    $"Passed: {results.PassedTests} " +
+                    $"Inconclusive: {results.InconclusiveTests} " +
+                    $"Failed: {results.FailedTests} " +
+                    $"Ignored: {results.SkippedTests}";
+                bundle.PutString(SummaryMessageArgumentName, message);
+                
+                bundle.PutLong(ExitCodeArgumentName, results.FailedTests == 0 ? 0 : 1);
+            };
+
+            entryPoint.Tests = Tests;
+
+            await entryPoint.RunAsync();
+
+            if (File.Exists(entryPoint.TestsResultsFinalPath))
+            {
+                bundle.PutString(ResultsPathArgumentName, entryPoint.TestsResultsFinalPath);
+            }
+            
+            if (bundle.GetLong(ExitCodeArgumentName, -1) == -1)
+            {
+                bundle.PutLong(ExitCodeArgumentName, 1);
+            }
+
+            Finish(Result.Ok, bundle);
+        }
+
+        class TestsEntryPoint : AndroidApplicationEntryPoint
+        {
+            readonly string resultsPath;
+            readonly string includeMethod;
+            readonly string excludeCategories;
+            readonly string excludeMethod;
+            readonly string includeClass;
+            readonly string excludeClass;
+
+            public TestsEntryPoint(Dictionary<string, string> arguments)
+            {
+                var root = Application.Context.GetExternalFilesDir(null)?.AbsolutePath;
+
+                var docsDir = Path.Combine(root, "Documents");
+
+                if (!Directory.Exists(docsDir))
+                {
+                    Directory.CreateDirectory(docsDir);
+                }
+
+                resultsPath = Path.Combine(docsDir, arguments[ResultsFileArgumentName]);
```

Comment:
Does this mean all of the different bundles will keep overwriting the same testResults file?

Reply:
Not necessary because we just provide default name and for each test different names could be provided from the parameter. But I may try to generate somehow the name from test info